### PR TITLE
[INLONG-10195][Sort] Hudi connector support audit ID

### LINKED ISSUE #10195: [Feature][Sort] Hudi connector support audit ID
### Description

Hudi connector support audit ID sent by the manager.

Parent issue is https://github.com/apache/inlong/issues/8803

### Use case

Hudi connector support audit ID.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
### Prepare a Pull Request

 - [INLONG-10195][Sort] Hudi connector support audit ID

- Fixes #10195 

- Parent issue: https://github.com/apache/inlong/issues/8803

### Motivation

Hudi connector not support audit function. So this pr will make it support audit function.

### Modifications

Modify Hudi connector sink.when sinkreceive data, it will send audit information at same time.Some implement are as follows:

Program will get the audit information from the incoming sql statement and pass it to the StreamWriteFunction class. During StreamWrite , the program will report the number of data received.
image

### CODE REVIEW

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/utils/Pipelines.java [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+     * and flushes the data set to disk.
+     *
+     * @param conf       The configuration
+     * @param dataStream The input data stream
+     * @return the stream write data stream pipeline
+     */
+    public static DataStream<Object> hoodieStreamWrite(Configuration conf, DataStream<HoodieRecord> dataStream,
+            MetricOption metricOption) {
+        if (OptionsResolver.isBucketIndexType(conf)) {
+            WriteOperatorFactory<HoodieRecord> operatorFactory = BucketStreamWriteOperator.getFactory(conf);
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/utils/Pipelines.java:331]: Miss metricOption here

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/utils/Pipelines.java]: fix it～， thx～

**Review by aloyszhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/bucket/BucketStreamWriteFunction.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+     *
+     * @param config The config options
+     */
+    public BucketStreamWriteFunction(Configuration config, MetricOption metricOption) {
+        super(config);
+        this.metricOption = metricOption;
+    }
+
+    @Override
+    public void open(Configuration parameters) throws IOException {
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/bucket/BucketStreamWriteFunction.java:87]: where is the IOException thrown out ?

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/StreamWriteFunction.java [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+     *
+     * @param config The config options
+     */
+    public StreamWriteFunction(Configuration config, MetricOption metricOption) {
+        super(config);
+        this.metricOption = metricOption;
+    }
+
+    @Override
+    public void open(Configuration parameters) throws IOException {
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/StreamWriteFunction.java:127]: ditto

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/bucket/BucketStreamWriteFunction.java (hunk 2 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/bucket/BucketStreamWriteFunction.java:87]: fixt it～ thx～

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/StreamWriteFunction.java (hunk 3 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hudi/src/main/java/org/apache/inlong/sort/hudi/sink/StreamWriteFunction.java]: fix it ～ thx～

**Review by aloyszhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**